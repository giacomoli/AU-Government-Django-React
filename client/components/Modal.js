import React from 'react'

export default ({ id, title, children, footer, size, bodyStyle = {} }) => {
  const modalSize = `modal-${size}`
  return (
    <div className="modal fade" tabIndex="-1" role="dialog" id={id}>
      <div className={`modal-dialog ${size ? modalSize : ''}`} role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title">{title}</h5>
            <button
              id="modalCloseButton"
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body" style={bodyStyle}>
            {children}
          </div>
          {footer && <div className="modal-footer">{footer}</div>}
        </div>
      </div>
      <style jsx>{`
        .close {
          cursor: pointer;
        }
      `}</style>
    </div>
  )
}
