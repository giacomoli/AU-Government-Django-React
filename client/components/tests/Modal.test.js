import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo';
import { reducer } from 'redux-form'
import { ApolloClient } from 'apollo-client';
import { createStore } from 'redux';
import Modal from '../Modal'

/* eslint-disable */
describe('Modal', () => {
  const client = new ApolloClient()
  const store = createStore(reducer)
  const initialValues = {
    id: 123,
    title: "Test Modal",
    children: <div>Test content</div>,
    footer: "Test Footer",
    size: 'lg',
    bodyStyle: {color: 'red'},
  }

  it('should render Modal without props', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <Modal
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('should render Modal with props', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <Modal
          id={initialValues.id}
          title={initialValues.title}
          children={initialValues.children}
          footer={initialValues.footer}
          size={initialValues.size}
          bodyStyle={initialValues.bodyStyle}
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
