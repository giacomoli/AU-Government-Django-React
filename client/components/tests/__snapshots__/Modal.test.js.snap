// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render Modal with props 1`] = `
<div
  className="jsx-2126699498 modal fade"
  id={123}
  role="dialog"
  tabIndex="-1"
>
  <div
    className="jsx-2126699498 modal-dialog modal-lg"
    role="document"
  >
    <div
      className="jsx-2126699498 modal-content"
    >
      <div
        className="jsx-2126699498 modal-header"
      >
        <h5
          className="jsx-2126699498 modal-title"
        >
          Test Modal
        </h5>
        <button
          aria-label="Close"
          className="jsx-2126699498 close"
          data-dismiss="modal"
          id="modalCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            className="jsx-2126699498"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="jsx-2126699498 modal-body"
        style={
          Object {
            "color": "red",
          }
        }
      >
        <div>
          Test content
        </div>
      </div>
      <div
        className="jsx-2126699498 modal-footer"
      >
        Test Footer
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal should render Modal without props 1`] = `
<div
  className="jsx-2126699498 modal fade"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="jsx-2126699498 modal-dialog "
    role="document"
  >
    <div
      className="jsx-2126699498 modal-content"
    >
      <div
        className="jsx-2126699498 modal-header"
      >
        <h5
          className="jsx-2126699498 modal-title"
        />
        <button
          aria-label="Close"
          className="jsx-2126699498 close"
          data-dismiss="modal"
          id="modalCloseButton"
          type="button"
        >
          <span
            aria-hidden="true"
            className="jsx-2126699498"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="jsx-2126699498 modal-body"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
